Style guide for german translations

tschweg opened the pull request:
As mentioned in #832 by @Xekyo , it could make sense to have a language-specific list of terms. 
This PR introduces a language-specific style guide based on the structure of the [original guide](STYLES.md), and supplements it where necessary with rules for German translations.

Open TODOs:
- [x] Agree on necessity, approach and structure of language-specific style guide
- [x] Review and extend section _Gebräuchliche Substantive_
- [x] Add content to section _Zusammengesetzte Begriffe_
- [x] Review and extend list of terms in section _Bevorzugte Begriffe_
 
@bitschmidty , @Xekyo , happy to get your input on this one.

Xekyo commented on STYLE-DE.md:
```diff
@@ -0,0 +1,68 @@
+der erste Buchstabe von gebräuchlichen Substantiven gross geschrieben.
+
+- bech32
+- bitcoin (die Währung)
```
Hätte ich groß geschrieben, wie wir auch 1 Euro, 1 Peso oder 1 US-Dollar groß schreiben.

tschweg replied:
Hatte ich klein geschrieben, wegen der speziellen Schreibweise im Englischen (Netzwerk vs. Währung). Technische Begriffe wie bech32 ebenfalls gross?

tschweg replied:
Fixed

Xekyo reviewed (commented):
Super, danke für den Anstoß dieses Dokuments. Planst du die alten Ausgaben durchzugehen um dortige Entscheidungen schon zu dokumentieren oder sollen wir uns das aufteilen? Sonst könnten wir auch einfach hier hinzufügen während Begriffe in zukünftigen Ausgaben relevant sind.

tschweg commented:
Ja, hatte vor die alten Ausgaben nochmal durchzugehen. Ich wollte mit dem WIP-PR zuerst mal die Diskussion bezüglich Struktur usw. anstossen. Wenn das Vorgehen so für Dich/Euch OK ist, füg ich mal noch etwas Content hinzu. Nach "Redaktionsschluss" würd ich dann die alten Ausgaben gemäss dem Styleguide geradeziehen.

Xekyo commented:
Ja schaut gut für mich aus. Denke mal, dass es auch für die anderen in Ordnung geht und festzuhalten was man entschieden hat scheint mir general ein guter Schritt um in der Zukunft Zeit zu sparen.

tschweg commented on STYLE-DE.md:
```diff
@@ -0,0 +1,68 @@
+Übersetzungen bevorzugt:
+
+_FIXME: Review and extend list of terms_
+| Ursprünglich| Bevorzugt | Zu vermeiden | Anmerkung |
```
Fixed

tschweg commented:
@Xekyo : I've added your suggestions and fixed the missing sections. I would propose to merge this one in and then build upon it with each subsequent newsletter/page that gets translated to German. Thoughts?
